addons/pos_discount/static/src/overrides/components/control_buttons/control_buttons.js

```diff
@@ -51,23 +53,21 @@ patch(ControlButtons.prototype, {
                 lines.filter((ll) => ll.isGlobalDiscountApplicable())
             );
 
+            const taxIds = this.pos.models["account.tax"].filter((tax) =>
+                tax_ids_array.includes(tax.id)
+            );
             // We add the price as manually set to avoid recomputation when changing customer.
             const discount = (-pc / 100.0) * baseToDiscount;
             if (discount < 0) {
                 order.add_product(product, {
                     price: discount,
                     lst_price: discount,
-                    tax_ids: tax_ids_array,
+                    tax_ids: taxIds,
```

Comment:
did you change how `add_product` accepts tax_ids? Does it accept objects now? From what I understand here, the tax_ids should be an array of numbers.